src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -156,17 +170,23 @@ public boolean handleInput(String input, boolean needsDetail, boolean inputValid
                 EntityScreen entityScreen = (EntityScreen)screen;
                 boolean autoLaunch = entityScreen.getAutoLaunchAction() != null && allowAutoLaunch;
                 addBreadcrumb = !autoLaunch;
-                if (input.startsWith("action ") || (autoLaunch) || !inputValidated) {
+                if (input.startsWith("action ") || (autoLaunch) || !inputValidated ||
+                        input.contentEquals(USE_SELECTED_VALUES)) {
```

Comment:
Is this necessary? I think if we ensure that this value is never saved in the `cacheSessionSelections` then it shouldn't be.

I also think that we currently are saving USE_SELECTED_VALUES into the cached session selections. I think line 317 below should change:

```diff
- restoreFactory.cacheSessionSelections(selections);
+ restoreFactory.cacheSessionSelections(menuSession.getSelections());
```

I'd be interested to see without that change whether `inputValidated` is correctly set for navigation after the selections are modified. I suspect it is not since we are not caching the GUID in the session selections.

Reply:
that's a good idea, agree that it's not caching the guid selection today in case wheh the selection array have `USE_SELECTED_VALUES`

Reply:
https://github.com/dimagi/formplayer/pull/1109/commits/bd2469675698752c88e4fc18a2286456ea5b0932